Add Pure Storage FlashBlade Integration (#1273)
DataDog/integrations-extras: purefb/README.md

# Agent Check: PureFB

## Overview

> This check monitors the [Pure Storage FlashBlade][3] through the [Datadog Agent][2] and the [Pure Storage FlashBlade OpenMetrics exporter][1]. 

Reviewer: Would it make more sense to link directly to the FlashBlade page in the docs for [3]? [https://www.purestorage.com/products/file-and-object/flashblade.html](https://www.purestorage.com/products/file-and-object/flashblade.html)

> This check monitors the [Pure Storage FlashBlade][3] through the [Datadog Agent][2] and the [Pure Storage FlashBlade OpenMetrics exporter][1]. 

Reviewer: Would it make more sense to link directly to the FlashBlade page in the docs for [3]? [https://www.purestorage.com/products/file-and-object/flashblade.html](https://www.purestorage.com/products/file-and-object/flashblade.html)

Author: Not at this time, we would like to link to the main product page.

The integration can provide performance data at the array, client, share and bucket level, as well as high-level capacity and configuration information.

You can monitor multiple FlashBlades and aggregate these into a single dashboard, or group them together by customer defined environment.

**This integration requires the following**:

 - FlashBlade Purity 3.2.x+
 - Agent v7.26.x+ to utilize OpenMetricsBaseCheckV2
 - Python 3

>  - The Pure Storage FlashBlade OpenMetrics exporter is installed and running in a containerized environment. Refer to the [GitHub repo][1] for installation instructions.

Reviewer: ```suggestion
 - The Pure Storage FlashBlade OpenMetrics exporter is installed and running in a containerized environment. Refer to the [Pure Storage GitHub repo][1] for installation instructions.
```

Author: updated.

## Setup

> Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the Autodiscovery Integration Templates for guidance on applying these instructions.

Reviewer: Does "Autodiscovery Integration Templates" need a link?

Author: updated with a link.

### Installation

1. [Download and launch the Datadog Agent][9].
2. Manually install the Pure FlashBlade integration. See [Use Community Integrations][10] for more details based on your environment.

#### Host

> To configure this check for an Agent running on a host, run `datadog-agent integration install -t datadog-purefb==<INTEGRATION_VERSION>`.

Reviewer: ```suggestion
To configure this check for an Agent running on a host, run `datadog-agent integration install -t datadog-purefb==1.0.0`.
```

Author: updated.

       - env:<env>
       - fb_array_name:<full_fqdn>
       - host:<full_fqdn>
    headers:
       Authorization: Bearer <api_token>
    min_collection_interval: 120

  - openmetrics_endpoint: http://<exporter_ip_or_fqdn>:<port>/metrics/clients?endpoint=<array_ip_or_fqdn>
    tags:
       - env:<env>
       - fb_array_name:<full_fqdn>
       - host:<full_fqdn>
    headers:
       Authorization: Bearer <api_token>
    min_collection_interval: 600

  - openmetrics_endpoint: http://<exporter_ip_or_fqdn>:<port>/metrics/usage?endpoint=<array_ip_or_fqdn>
    tags:
       - env:<env>
       - fb_array_name:<full_fqdn>
       - host:<full_fqdn>
    headers:
       Authorization: Bearer <api_token>
    min_collection_interval: 600

```

2. [Restart the Agent][5].

### Validation

[Run the Agent's status subcommand][6] and look for `purefb` under the Checks section.

### Troubleshooting

#### Arrays are not showing in dashboard

The dashboards included in this integration use the tags `env`, `host`, and `fb_array_name`. Make sure that these are set per instance.

```yaml
 tags:
    - env:<env>
    - fb_array_name:<full_fqdn>
    - host:<full_fqdn>
```

#### Increasing collection interval

> For the `/array` endpoint The Pure Storage FlashBlade check sets `min_collection_interval` to `120` by default , and the minimum recommended value is `15`. You may increase/decrease `min_collection_interval` in the `purefb.d/conf.yaml` file if necessary:

Reviewer: ```suggestion
For the `/array` endpoint, the Pure Storage FlashBlade check sets `min_collection_interval` to `120` by default, and the minimum recommended value is `15`. You may increase/decrease `min_collection_interval` in the `purefb.d/conf.yaml` file if necessary:
```

Author: updated.

```yaml
min_collection_interval: 120
```

> For the `/clients`, and `/usage` endpoints The Pure Storage FlashBlade check sets `min_collection_interval` to `600` by default , and the minimum recommended value is `120`. You may increase/decrease `min_collection_interval` in the `purefb.d/conf.yaml` file if necessary:

Reviewer: ```suggestion
For the `/clients`, and `/usage` endpoints, the Pure Storage FlashBlade check sets `min_collection_interval` to `600` by default , and the minimum recommended value is `120`. You may increase/decrease `min_collection_interval` in the `purefb.d/conf.yaml` file if necessary:
```

Author: updated.

# Agent Check: PureFB

## Overview

This check monitors the [Pure Storage FlashBlade][3] through the [Datadog Agent][2] and the [Pure Storage FlashBlade OpenMetrics exporter][1]. 

> The integration can provide performance data at the array, client, share and bucket level, as well as high-level capacity and configuration information.

Reviewer 2: ```suggestion
The integration can provide performance data at the array, client, share, and bucket level, as well as high-level capacity and configuration information.
```

> You can monitor multiple FlashBlades and aggregate these into a single dashboard, or group them together by customer defined environment.

Reviewer 2: ```suggestion
You can monitor multiple FlashBlades and aggregate these into a single dashboard, or group them together by customer-defined environment.
```

**This integration requires the following**:

 - FlashBlade Purity 3.2.x+

>  - Agent v7.26.x+ to utilize OpenMetricsBaseCheckV2

Reviewer 2: ```suggestion
 - Datadog Agent v7.26.x+ to use OpenMetricsBaseCheckV2
```

 - Python 3
 - The Pure Storage FlashBlade OpenMetrics exporter is installed and running in a containerized environment. Refer to the [Pure Storage GitHub repo][1] for installation instructions.

## Setup

Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the [Autodiscovery Integration Templates][13] for guidance on applying these instructions.

### Installation

1. [Download and launch the Datadog Agent][9].
2. Manually install the Pure FlashBlade integration. See [Use Community Integrations][10] for more details based on your environment.

#### Host

To configure this check for an Agent running on a host, run `datadog-agent integration install -t datadog-purefb==1.0.0`.

### Configuration

1. Create a user on your FlashBlade with the Read-Only role and generate an API token for this user.

> 2. Add the following configuration block to the `purefb.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your PureFB performance data. See the sample [purefb.d/conf.yaml][4] for all available configuration options.

Reviewer 2: ```suggestion
2. Add the following configuration block to the `purefb.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory, to start collecting your PureFB performance data. See the sample [purefb.d/conf.yaml][4] for all available configuration options.
```

       - env:<env>
       - fb_array_name:<full_fqdn>
       - host:<full_fqdn>
    headers:
       Authorization: Bearer <api_token>
    min_collection_interval: 120

  - openmetrics_endpoint: http://<exporter_ip_or_fqdn>:<port>/metrics/clients?endpoint=<array_ip_or_fqdn>
    tags:
       - env:<env>
       - fb_array_name:<full_fqdn>
       - host:<full_fqdn>
    headers:
       Authorization: Bearer <api_token>
    min_collection_interval: 600

  - openmetrics_endpoint: http://<exporter_ip_or_fqdn>:<port>/metrics/usage?endpoint=<array_ip_or_fqdn>
    tags:
       - env:<env>
       - fb_array_name:<full_fqdn>
       - host:<full_fqdn>
    headers:
       Authorization: Bearer <api_token>
    min_collection_interval: 600

```

2. [Restart the Agent][5].

### Validation

[Run the Agent's status subcommand][6] and look for `purefb` under the Checks section.

### Troubleshooting

#### Arrays are not showing in dashboard

The dashboards included in this integration use the tags `env`, `host`, and `fb_array_name`. Make sure that these are set per instance.

```yaml
 tags:
    - env:<env>
    - fb_array_name:<full_fqdn>
    - host:<full_fqdn>
```

#### Increasing collection interval

> For the `/array` endpoint, the Pure Storage FlashBlade check sets `min_collection_interval` to `120` by default, and the minimum recommended value is `15`. You may increase/decrease `min_collection_interval` in the `purefb.d/conf.yaml` file if necessary:

Reviewer 2: ```suggestion
For the `/array` endpoint, the Pure Storage FlashBlade check sets `min_collection_interval` to `120` by default, and the minimum recommended value is `15`. You may increase or decrease `min_collection_interval` in the `purefb.d/conf.yaml` file if necessary:
```

Reviewer 2: For clarity, as well as to facilitate the translation process for Datadog docs

```yaml
min_collection_interval: 120
```

> For the `/clients`, and `/usage` endpoints, the Pure Storage FlashBlade check sets `min_collection_interval` to `600` by default , and the minimum recommended value is `120`. You may increase/decrease `min_collection_interval` in the `purefb.d/conf.yaml` file if necessary:

Reviewer 2: ```suggestion
For the `/clients`, and `/usage` endpoints, the Pure Storage FlashBlade check sets `min_collection_interval` to `600` by default , and the minimum recommended value is `120`. You may increase or decrease `min_collection_interval` in the `purefb.d/conf.yaml` file if necessary:
```

Reviewer 2: As above
